src/System.Text.Primitives/System/Text/Encoders/Utf8Validator.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+
+namespace System.Buffers.Text
+{
+    /// <summary>
+    /// Allows processing discontiguous buffers of UTF-8 data, letting the caller know whether
+    /// any data seen so far is invalid UTF-8. The caller should invoke <see cref="TryConsume(ReadOnlySpan{byte, bool})"/>
+    /// in a loop until all data has been consumed.
+    /// </summary>
+    /// <remarks>
+    /// This type is a mutable struct.
+    /// </remarks>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public struct Utf8Validator
```

Comment:
I feel like we have too many types. Developers will have to remember where each of the API lives. Could we have all Utf8 related methods in the Utf8 static class?